### PR TITLE
Update user guide based on latest features and ensure consistency

### PR DESCRIPTION
**Pull request by techjay-c**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -241,15 +273,34 @@ This also means that you can use `t/` to remove all tags from a dentist using th
 
 ## Part 2 - Patient Features
 
-### Adding a patient: `add-patient`
-
+### Adding a patient : `add-patient`
```
Comment (ruth-lim) [inline docs/UserGuide.md:276]: Are we standardising to have a space before the colon? If so, do change for the dentist features above :)

File: docs/UserGuide.md [hunk 2]
```diff
@@ -241,15 +273,34 @@ This also means that you can use `t/` to remove all tags from a dentist using th
...
+**:information_source: Notes about adding a patient without email and address:**<br>
+ToothTracker will put the <code>EMAIL</code> as "NoEmailProvided@ToBeAdded.com"
+and put the <code>ADDRESS</code> as "No Address Provided." by default.
+
+If you wish to update them at a later time, you can use <code>edit-patient</code> to edit them.
+</div>
+
+**Examples:**
 
 * `add-patient n/John p/90676622 b/06-06-1998 g/M r/Allergic to Peanuts tr/Braces e/johntan@gmail.com h/60 Harvey Avenue t/Urgent` <br>
```
Comment (ruth-lim) [inline docs/UserGuide.md:305]: I think it should requesting for Braces treatment, not cleaning treatment.

Comment (ruth-lim) [reply docs/UserGuide.md:305]: Also, should the explanation of the example be written by following the order of the patient format?

**Review by qyaner — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 3]
```diff
@@ -279,60 +334,79 @@ Deletes the patient with the specified `PATIENT_ID` from ToothTracker.
...
-* `search-patient 88` searches for a patient with the ID `88`. If one or more matching patients are found based on
-  your search criteria, the system will list the matching patients along with their details.
+Search for patients by matching names with a keyword. This command helps you find patient records that match your search criteria.
+This is useful when:
+
+* You need to find patients whose names match your search criteria.
+
+**Format:** `search-patient KEYWORD`
+
+**Important Notes for searching dentists by keyword:**
```
Comment (ruth-lim) [inline docs/UserGuide.md:378]: should be "patients" not "dentists"

File: docs/UserGuide.md [hunk 4]
```diff
@@ -343,18 +417,18 @@ Edits one or more details of the patient at the specified patient ID. This is he
 
 **Format:** `PATIENT_ID [n/NAME] [p/PHONE] [b/BIRTHDATE] [g/GENDER] [r/REMARK] [tr/TREATMENT] [e/EMAIL] [h/ADDRESS] [t/TAG]`
 
-Examples:
+**Examples:**
 * `edit-patient 1 p/91234567 e/johndoe@example.com r/Allergic to Peanuts` updates the 
-phone number, email address and remarks of patient with Patient ID of 1
+phone number, email address and remarks of the patient with `PATIENT_ID 1`
```
Comment (ruth-lim) [inline docs/UserGuide.md:422]: should be "`PATIENT_ID` 1".

File: docs/UserGuide.md [hunk 5]
```diff
@@ -343,18 +417,18 @@ Edits one or more details of the patient at the specified patient ID. This is he
...
 
-* `edit-patient 2  n/Alex` updates the name of patient with ID of 2 to Alex
+* `edit-patient 2  n/Alex` updates the name of the patient with `PATIENT_ID 2` to Alex
```
Comment (ruth-lim) [inline docs/UserGuide.md:424]: should be "`PATIENT_ID` 2".

File: docs/UserGuide.md [hunk 6]
```diff
@@ -426,43 +500,41 @@ This is helpful when:
...
-The `DENTIST_ID` refers to the index number shown in the displayed list of dentists. <br>
-You may use `list-dentist` to check out the dentist's ID first.
+`DENTIST_ID` refers to the index number shown in the displayed list of dentists. <br>
+You may use `list-dentist` to check out the `DENTIST_ID` first.
 </div>
 
+**Example:**
 
-Example:
+* `filter-appointment dentist 1` returns all appointments under the dentist with `DENTIST_ID 1`.
```
Comment (ruth-lim) [inline docs/UserGuide.md:519]: should be "`DENTIST_ID` 1".

File: docs/UserGuide.md [hunk 7]
```diff
@@ -426,43 +500,41 @@ This is helpful when:
...
+`PATIENT_ID` refers to the index number shown in the displayed list of patients. <br>
+You may use `list-patient` to check out the `PATIENT_ID` first.
 </div>
 
+**Example:**
 
-Example:
-
-* `filter-appointment patient 1` returns all appointments under the patient with the patient ID 1.
+* `filter-appointment patient 1` returns all appointments under the patient with the `PATIENT_ID 1`.
```
Comment (ruth-lim) [inline docs/UserGuide.md:537]: should be "`PATIENT_ID` 1"

**Pull request merged**